### PR TITLE
Fixes all warnings when publishing

### PR DESCRIPTION
**Pull request by vegaro**
I ran a dry-run of the publish command and it showed some warnings. I am adding the specific warnings as comments in this PR

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 1. Update to the latest SDK versions in purchases_flutter.podspec and build.gradle.
 1. Run `flutter format .`
 1. Update versions in VERSIONS.md.
-1. Update CHANGELOG.latest.md with the changes for the current version (to be used by Fastlane for the github release notes)
+1. Update CHANGELOG-LATEST.md with the changes for the current version (to be used by Fastlane for the github release notes)
```
Comment (vegaro) [inline RELEASING.md:5]: It was telling me CHANGELOG.latest.md should be named CHANGELOG.md

File: lib/offering_wrapper.dart [hunk 2]
```diff
@@ -58,8 +58,8 @@ class Offering {
   /// Retrieves a specific package by identifier, use this to access custom
   /// package types configured in the RevenueCat dashboard.
   Package? getPackage(String identifier) {
-    return availablePackages.firstWhereOrNull(
-        (package) => package.identifier == identifier);
+    return availablePackages
+        .firstWhereOrNull((package) => package.identifier == identifier);
```
Comment (vegaro) [inline lib/offering_wrapper.dart:62]: Result of `flutter format .`

File: lib/entitlement_info_wrapper.dart [hunk 3]
```diff
@@ -134,7 +134,6 @@ enum PeriodType {
 
   /// If the period type couldn't be determined.
   unknown
-
```
Comment (vegaro) [inline lib/entitlement_info_wrapper.dart:137]: Result of `flutter format .`

File: pubspec.yaml [hunk 4]
```diff
@@ -8,12 +8,12 @@ documentation: https://docs.revenuecat.com/docs
 
 environment:
   sdk: '>=2.12.0 <3.0.0'
-  flutter: ">=1.12.13+hotfix.6 <2.0.0"
```
Comment (vegaro) [inline pubspec.yaml:11]: * The Flutter constraint should not have an upper bound.
  In your pubspec.yaml the constraint is currently `>=1.12.13+hotfix.6 <2.0.0`.
  
  You can replace that with just the lower bound: `>=1.12.13+hotfix.6`.
  
  See https://dart.dev/go/flutter-upper-bound-deprecation

File: pubspec.yaml [hunk 5]
```diff
@@ -8,12 +8,12 @@ documentation: https://docs.revenuecat.com/docs
...
+  flutter: ">=1.12.13+hotfix.6"
 
 dependencies:
   flutter:
     sdk: flutter
-  collection: ^1.15.0-nullsafety.4
+  collection: ^1.15.0
```
Comment (vegaro) [inline pubspec.yaml:16]: This is already a final release so we can update it. The warning was:

* Packages dependent on a pre-release of another package should themselves be published as a pre-release version. If this package needs collection version 1.15.0-nullsafety.4, consider publishing the package as a pre-release instead.
  See https://dart.dev/tools/pub/publishing#publishing-prereleases For more information on pre-releases.

File: test/entitlement_info_test.dart [hunk 6]
```diff
@@ -43,9 +43,9 @@ void main() {
...
       "billingIssueDetectedAt": null,
-      "billingIssueDetectedAtMillis": null};
-    final entitlementInfo =
-        EntitlementInfo.fromJson(entitlementInfoJson);
+      "billingIssueDetectedAtMillis": null
+    };
+    final entitlementInfo = EntitlementInfo.fromJson(entitlementInfoJson);
 
     expect(entitlementInfo.store, Store.unknownStore);
   });
```
Comment (vegaro) [inline test/entitlement_info_test.dart:51]: Result of formatting

**Review by aboedo — APPROVED** [review]

**Pull request merged**